### ISSUE TITLE
[Question] Best practices to change harvester-mgmt interface IP netmask

### ISSUE THREAD

**Issue by zp4rand0miz31** [harvester/harvester#2823]
Hi,

We migrated our main LAN network from a 10.20.0.0/24 network to a 10.20.0.0/16 network.

To achieve this migration, we patched the /oem/99_custom.yaml file so that the netmask is adjusted to NETMASK=255.255.0.0. 
Our DHCP server properly broadcast the netmask 255.255.0.0, even if not used in this configuration as all the nodes are configured statically.
We rebooted all the nodes.

Since then we observed the following:

- sometimes, nodes can't reach each other. We tracked back this problem to a route added to some hosts routing table : 
route is "10.20.0.0/24 via 10.20.0.1 dev harvester-mgmt proto kernel". This route seems to be added after a while, when rke2-agent is ran (periodically).
- after this, we lose the virtual IP address of the kube api-server, which finally renders the whole cluster as unavailable.  

I am not sure if any configuration remains inside the cluster and whether netmask for host IP addresses are saved. 

I would expect this has to be done via kubeadm, but I am not sure how to use this command with a harvester cluster. More, I did not find lot of configurations files related to IP netmask (which is kind of logical, as nodes IP addresses are not managed by the cluster itself).

Any ideas on what we do wrong here  ? Is there some configuration with calico to be updated ? 
Thanks.

**Comment by bk201** [comment]
@zp4rand0miz31 Thanks for reporting. What's your [VIP setting](https://docs.harvesterhci.io/v1.0/install/harvester-configuration/#installvip)? Is it a static one or a DHCP one?

**Comment by zp4rand0miz31** [comment]
> @zp4rand0miz31 Thanks for reporting. What's your [VIP setting](https://docs.harvesterhci.io/v1.0/install/harvester-configuration/#installvip)? Is it a static one or a DHCP one?

Checking the inital config in /oem/99_custom.yaml, from the ManagedChart 'harvester':
```
 service:
       vip: 
             enabled: true
             mode: "static"
             ip: "10.20.0.76"
             hwAddress: ""
```

We do have the same values in harvester.config. The live config in the fleet-local namespace is the same too.

Some precisions on my request:

after boot, we have the following routing table (i removed the calico related routes):
```
default via 10.20.0.1 dev harvester-mgmt 
10.20.0.0/16 dev harvester-mgmt proto kernel scope link src 10.20.0.70 
```
which seems legit. 

I am not sure whether the bad route "10.20.0.0/2'" is added by rke2 or rancher-system-agent.

Kind regards, 
Thanks.

**Comment by bk201** [comment]
@yaocw2020 @tjjh89017 Any insights on potential causes? Thanks.

**Comment by tjjh89017** [comment]
> sometimes, nodes can't reach each other. We tracked back this problem to a route added to some hosts routing table :
route is "10.20.0.0/24 via 10.20.0.1 dev harvester-mgmt proto kernel". This route seems to be added after a while, when rke2-agent is ran (periodically).

Could you help us to check the DHCP server again?
v1.0.x network controller will ask DHCP for subnet info and add route to some master node for that.
So if you have VLAN network in 10.20.0.0/16 subnet and DHCP server is still announce /24 subnet, network controller will setup a /24 route for it.

**Comment by zp4rand0miz31** [comment]
> Could you help us to check the DHCP server again?

I checked the configuration of our DHCP server, it distributes properly IPs on /16. I also captured DHCP request/offer packets on both `harvester-br0` and `harvester-mgmt` interfaces, and I can confirm that the option 11 (Subnet-Mask) is 255.255.0.0 in DHCP offer packets.

Does the network controller intercepts outbound and interpret (from the cluster point of view) DHCP requests ? Our VM images were prepared on a different network (the one proposed by packer by default) and could ask for a RENEW DHCP that could eventually trigger the route addition ?  I did not confirmed this theory yet.

**Comment by tjjh89017** [comment]
> > Could you help us to check the DHCP server again?
> 
> I checked the configuration of our DHCP server, it broadcasts properly on /16. I also captured DHCP request/offer packets on both `harvester-br0` and `harvester-mgmt` interfaces, and I can confirm that the option 11 (Subnet-Mask) is 255.255.0.0 in DHCP offer packets.
> 
> Does the network controller intercepts outbound and interpret (from the cluster point of view) DHCP requests ? Our VM images were prepared on a different network (the one proposed by packer by default) and could ask for a RENEW DHCP that could eventually trigger the route addition ? I did not confirmed this theory yet.

Network controller will start a DHCP client pod and attach to every net-attach-def.
Try to grab DHCP info and setup route in nodes.

But I'm not sure the reason, it will setup /24 route if your DHCP server is /16 and correct.

cc @yaocw2020

**Comment by zp4rand0miz31** [comment]
> Network controller will start a DHCP client pod and attach to every net-attach-def.
> Try to grab DHCP info and setup route in nodes.

Hi, not sure whether you are waiting on some action from me. I don't know how to perform the described action.

>But I'm not sure the reason, it will setup /24 route if your DHCP server is /16 and 
Looks like a bug, yeah ?

Thanks.

**Comment by tjjh89017** [comment]
> > Network controller will start a DHCP client pod and attach to every net-attach-def.
> > Try to grab DHCP info and setup route in nodes.
> 
> Hi, not sure whether you are waiting on some action from me. I don't know how to perform the described action.

No, I just describe the behavior or network controller.
but could you try v1.1.0 when we release v1.1.0?
We have a refactor for network function
thanks a lot

**Comment by zp4rand0miz31** [comment]
Hi, 
Follow up on this:
- we installed a new cluster based on 1.0.3 version and specified a /16 network right from the start
- no spurious /24 routes were added when creating VMs or other events

However, not related to this problem, we hit another problem when adding a 3rd node to the cluster (seems bug #2736, fixed in 1.1.0). We will try upgrading from 1.0.3 to 1.1.0, I hope this will be seamless ! (Previous upgrades attempts from 1.0.2 to 1.0.3 failed).   

1.1.0 is due tomorrow ?

Kind regards guys,
